firebase-firestore/src/androidTest/java/com/google/firebase/firestore/FirestoreTest.java

```diff
@@ -969,11 +967,9 @@ public void testMaintainsPersistenceAfterRestarting() {
     DocumentReference docRef = testDocument();
     FirebaseFirestore firestore = docRef.getFirestore();
     waitFor(docRef.set(map("foo", "bar")));
-    Context context = InstrumentationRegistry.getContext();
 
     waitFor(AccessHelper.shutdown(firestore));
-    FirebaseApp app2 = FirebaseApp.initializeApp(context);
-    FirebaseFirestore firestore2 = FirebaseFirestore.getInstance(app2);
+    FirebaseFirestore firestore2 = testFirestore();
```

Comment:
This change breaks the intent of the test.

`testFirestore()` creates a new Firestore instance associated with a new Firestore app. This guarantees that we'll use a different persistence key that the one we actually use. This test will pass even if clearPersistence is unimplemented.

The purpose of what we're doing here is attempting to create a new Firestore instance associated with the same app as the one before.

Reply:
> This test will pass even if clearPersistence is unimplemented.

Hmm... you're right. However, the *original* tests will also pass with an effectively empty clearPersistence. i.e.:
```
$ git diff
diff --git a/firebase-firestore/src/main/java/com/google/firebase/firestore/FirebaseFirestore.java b/firebase-firestore/src/main/java/com/google/firebase/firestore/FirebaseFirestore.java
index e935416..5021e17 100644
--- a/firebase-firestore/src/main/java/com/google/firebase/firestore/FirebaseFirestore.java
+++ b/firebase-firestore/src/main/java/com/google/firebase/firestore/FirebaseFirestore.java
@@ -404,17 +404,8 @@ public class FirebaseFirestore {
     if (this.clientRunning) {
       throw new IllegalStateException("Persistence cannot be cleared while the client is running.");
     }
-    final TaskCompletionSource<Void> source = new TaskCompletionSource<>();
-    asyncQueue.enqueueAndForget(
-        () -> {
-          try {
-            SQLitePersistence.clearPersistence(context, databaseId, persistenceKey);
-            source.setResult(null);
-          } catch (FirebaseFirestoreException e) {
-            source.setException(e);
-          }
-        });
-    return source.getTask();
+
+    return Tasks.forResult(null);
   }
 
   FirestoreClient getClient() {
$ ANDROID_HOME=~/Android/Sdk ./gradlew :firebase-firestore:connectedCheck
<successful output here>
```

I'll look into this a bit more tomorrow.

Reply:
Ok; I've reworked the tests. They now:
a) Properly fail when I noop clearPersistence (i.e. when I apply the above patch)
b) Pass within g3.